### PR TITLE
Implement onEndReached and onEndThreshold for reservation-list on agenda

### PR DESCRIPTION
**Pull request by geraldlee2001**
added onEndReached props and onEndThreshold props for reservation-list

### CODE REVIEW

**Review by nityanantan23 — APPROVED** [review]

**Review by nityanantan — APPROVED** [review]

File: ios/Podfile.lock [hunk 1]
```diff
@@ -379,7 +379,7 @@ SPEC CHECKSUMS:
   boost-for-react-native: 39c7adb57c4e60d6c5479dd8623128eb5b3f0f2c
   DoubleConversion: cf9b38bf0b2d048436d9a82ad2abe1404f11e7de
   FBLazyVector: e686045572151edef46010a6f819ade377dfeb4b
-  FBReactNativeSpec: 92869e54fbef651850edc0f8ad2e5e53421f5fbe
+  FBReactNativeSpec: b6b72b2bd8b6908be96832fe499865cc66897e86
```
Comment (Inbal-Tish) [inline ios/Podfile.lock:382]: this file should not be pushed. Please remove

File: package.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "name": "react-native-calendars",
   "version": "1.22.0",
   "main": "src/index.js",
+  "types":"src/agenda.d.ts",
```
Comment (Inbal-Tish) [inline package.json:5]: Please remove

File: src/agenda/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+import { Animated, LayoutChangeEvent, NativeSyntheticEvent, NativeScrollEvent } from 'react-native';
+import { DateData } from '../types';
+import { CalendarListProps } from '../calendar-list';
+import ReservationList, { ReservationListProps } from './reservation-list';
+export declare type ReservationItemType = {
+    name: string;
+    height: number;
+    day: XDate;
+};
+export declare type ReservationsType = {
```
Comment (Inbal-Tish) [inline src/agenda/index.d.ts:13]: please remove d.ts files

File: src/agenda/index.js [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+const HEADER_HEIGHT = 104;
+const KNOB_HEIGHT = 24;
+/**
+ * @description: Agenda component
+ * @extends: CalendarList
+ * @extendslink: docs/CalendarList
+ * @example: https://github.com/wix/react-native-calendars/blob/master/example/src/screens/agenda.js
+ * @gif: https://github.com/wix/react-native-calendars/blob/master/demo/agenda.gif
+ */
+export default class Agenda extends Component {
```
Comment (Inbal-Tish) [inline src/agenda/index.js:29]: agenda is already a tsx file. You need to fix this

**Comment by Inbal-Tish** [comment]
@geraldlee2001 Thank you for your PR. Please remove all generated file and keep only the relevant ones so I can review this.

**Pull request closed without merging**